### ISSUE TITLE
"Key too large: 4104 > 4096" when registering account key with ACME server

### ISSUE THREAD

**Issue by ixdy** [lukas2511/dehydrated#282]
While calling `nfsn-setup.sh` from https://github.com/nikjft/lets-nfsn.sh (which uses dehydrated):

```
Submodule path 'dehydrated': checked out '319852dc8cf167d8baeb0848b4998e86a02aa2d0'
 + Generating configuration...
 + Generating fallback configuration...
 + Installing hook script...
 + Generating domains.txt...
 + Performing initial run...
# INFO: Using main config file /home/private/lets-nfsn.sh/dehydrated/config
+ Generating account key...
+ Registering account key with ACME server...
  + ERROR: An error occurred while sending post-request to https://acme-v01.api.letsencrypt.org/acme/new-reg (Status 400)

Details:
{
  "type": "urn:acme:error:malformed",
  "detail": "Key too large: 4104 \u003e 4096",
  "status": 400
}



Error registering account key. See message above for more information.
```

Retrying doesn't help - the key is always 8 bytes too long.

**Comment by ixdy** [comment]
Reverting 319852dc8cf167d8baeb0848b4998e86a02aa2d0 fixed this.

**Comment by lukas2511** [comment]
@ixdy Mh... weird... that change shouldn't really make it longer... if any it should be shorter... What OS are you running this on?

**Comment by ixdy** [comment]
This is FreeBSD 10.3-RELEASE-p6.

**Comment by vsatmydynipnet** [comment]
Same error here: FreeBSD 10.3-RELEASE-p7.

**Comment by bhagerty** [comment]
Same error here, FreeBSD 10-3-RELEASE-p6. Any plan for solving this? What about reverting the change @ixdy pointed out?

**Comment by rowanthorpe** [comment]
I used the below code locally to keep the "trim leading nulls" logic in a more portable way (including FreeBSD's sed - which doesn't like `\x00`, or semicolons between commands, or branch-commands to not-yet-declared labels, for that matter). If you think this is not too ugly to use, let me know and I'll send it as a pull request. It requires `od` which seems to be effectively universally available, and with `-t x1` to avoid any endianness issues.

```
urlbase64() {
    printf -- "$(od -t x1 -An -v | tr -d '\t\r\n ' | _sed -e 's/(^|0a)(00)+/\1/g' -e 's/(.{2})/\\x\1/g')" | \
        openssl base64 -e | tr -d '\r\n' | _sed -e 's:=*$::g' -e 'y:+/:-_:'
}
```

**Comment by lukas2511** [comment]
Closing this ticket as this was solved by reverting the changes for #275.
